manifests/init.pp

```diff
@@ -249,4 +254,14 @@
   class { 'candlepin::artemis': } ~>
   class { "candlepin::database::${candlepin::db_type}": } ~>
   Class['candlepin::service']
+
+  # Add the candlepin.consumer.facts.match_regex option if provided
+  if $facts_match_regex {
```

Comment:
Drop this whole block you added, its not needed, and I think you'll find the tests should pass.